### DISCUSSION TITLE
Api documentation using own external yaml or Json file.

### DISCUSSION THREAD

**Discussion by rohit151990** [fastapi/fastapi#6522] [Questions] [answered]
Hi Friends,

I created my own swagger file and that I want to give for my api definition. I tried with given approach "app = FastAPI(openapi_url="/openapi.json")" but it is not loading my generated json file.

Please suggest what should I do so that I can use my own created swagger file in fastapi.

Regards,

**Comment by rohit151990** [comment, marked as answer]
Thanks Guys I got the solution.
this is how I can do this.

from fastapi import FastAPI
import json

app = FastAPI()

def custom_openapi():
    with open("openapi.json", "r") as openapi:
        return json.load(openapi)

app.openapi = custom_openapi

**Comment by burceam-homex** [comment]
I've tried this, and it's not working for me. I'm trying the same thing: I manually wrote a custom yaml/json file, and I'd like to serve this custom file instead of the auto-generated documentation.
Here is my code:
```
app = FastAPI()
def custom_openapi():
    with open("my_openapi.json", "r") as openapi:
        return json.load(openapi)
app.openapi = custom_openapi
```
With all this, if I go to `localhost:8080/docs`, or `localhost:8080/openapi.json`, I get the default, auto-generated documentation. If I explicitly go to `localhost:8080/my_openapi.json`, I get a 404 Not Found.
Would appreciate any help. Thank you!

Edit: eventually this did work. My problem was not copying the specific subfolder via the Dockerfile.
